### PR TITLE
MINOR: Improve close tests of caching state store

### PR DESCRIPTION
**Pull request by cadonna**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/state/internals/CachingKeyValueStoreTest.java [hunk 1]
```diff
@@ -122,38 +123,34 @@ public void shouldAvoidFlushingDeletionsWithoutDirtyKeys() {
...
         EasyMock.reset(underlyingStore);
         underlyingStore.close();
         EasyMock.replay(underlyingStore);
 
-        try {
-            store.close();
-        } catch (final RuntimeException exception) {
-            EasyMock.verify(underlyingStore);
-        }
+        assertThrows(RuntimeException.class, store::close);
```
Comment (cadonna) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/CachingKeyValueStoreTest.java:136]: I added the check for the exception.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/CachingKeyValueStoreTest.java [hunk 2]
```diff
@@ -122,38 +123,34 @@ public void shouldAvoidFlushingDeletionsWithoutDirtyKeys() {
...
+        EasyMock.verify(cache, underlyingStore);
```
Comment (cadonna) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/CachingKeyValueStoreTest.java:137]: I added the verification for the call to `close()` of underlying store and cache.

**Comment by cadonna** [comment]
Call for review: @vvcephei @mjsax

**Review by mjsax — APPROVED** [review]
LGTM.

**Comment by mjsax** [comment]
Retest this please

**Comment by mjsax** [comment]
Java 8: `org.apache.kafka.streams.integration.QueryableStateIntegrationTest.concurrentAccesses`
Java 11 timed out.

Retest this please.

**Review by vvcephei — APPROVED** [review]
Thanks! (And sorry again for mucking about so much with your last PR)

**Comment by vvcephei** [comment]
retest this please

**Comment by vvcephei** [comment]
The last run had the same unrelated failure on `org.apache.kafka.streams.integration.QueryableStateIntegrationTest.concurrentAccesses` for java 8, and also had like every broker api test failed on java 11.

I feel pretty good about merging it, WDYT, @mjsax ?

**Comment by mjsax** [comment]
Up to you :)

**Review by andrewchoi5 — APPROVED** [review]

**Comment by vvcephei** [comment]
That time, Jenkins shut down in the middle of the test. I'll run the tests locally and merge if passing.

**Comment by vvcephei** [comment]
Passed.
```
$ ./gradlew clean systemTestLibs :streams:test :streams:test-util:test
...
BUILD SUCCESSFUL in 12m 21s
```

**Pull request merged**